packages/web/src/components/search/SearchBox.js

```diff
@@ -506,9 +518,6 @@ const SearchBox = (props) => {
 		if (value === undefined) {
 			setValue(inputValue, false, props, undefined, true, false);
 		} else if (onChange) {
-			// handle caret position in controlled components
-			handleCaretPosition(e);
```

Comment:
This didn't have any effect and the cursor still shifted to the end. I suspect that when react version is `v17` or lower then only this works.